src/query/ts/m3db/convert.go

```diff
@@ -54,31 +57,56 @@ func (b blockReplicas) Less(i, j int) bool {
 	return b[i].start.Before(b[j].start)
 }
 
-// ConvertM3DBSeriesIterators converts m3db SeriesIterators to SeriesBlocks
-// which are used to construct Blocks for query processing.
-func ConvertM3DBSeriesIterators(iterators encoding.SeriesIterators, iterAlloc encoding.ReaderIteratorAllocate) ([]SeriesBlocks, error) {
-	defer iterators.Close()
-	multiSeriesBlocks := make([]SeriesBlocks, iterators.Len())
+// ConvertM3DBSeriesIterators converts series iterators to iterator blocks
+func ConvertM3DBSeriesIterators(
+	iterators encoding.SeriesIterators,
+	tagOptions models.TagOptions,
+	bounds *models.Bounds,
+) []block.MultipurposeBlock {
+	return []block.MultipurposeBlock{
+		MakeEncodedBlock(
+			iterators.Iters(),
+			bounds,
+			tagOptions,
+			true,
+		),
+	}
+}
 
-	for i, seriesIterator := range iterators.Iters() {
-		blockReplicas, err := blockReplicasFromSeriesIterator(seriesIterator, iterAlloc)
+// ConvertM3DBSegmentedBlockIterators converts series iterators to a list of blocks
+func ConvertM3DBSegmentedBlockIterators(
+	iterators encoding.SeriesIterators,
+	iterAlloc encoding.ReaderIteratorAllocate,
+	tagOptions models.TagOptions,
+	bounds *models.Bounds,
+	pools encoding.IteratorPools,
+) ([]block.MultipurposeBlock, error) {
+	return nil, fmt.Errorf("WIP: splitting blocks will require lookback propagation")
```

Comment:
this is a bit confusing. Is the code below not needed at all ?

Reply:
The below code is fine for lookback = 0s; however there will need to be changes to update it to be able to handle lookback times since blocks in a series will need to communicate with other blocks in that series to provide lookback values.

Reply:
there's no way to test the code below :( Maybe move this out into a different function and have a test ?

Reply:
It's not currently in use so no tests are necessary; this is the code required to transform a series iterator into series iterator blocks

Reply:
let's just comment it out then ?

Reply:
It gets errored out already

Reply:
Some linters complain about unreachable code. Usually, it's not a great practice to keep unreachable code around, just in case you don't clean it up later. I'm not sure what's your hesitation here. I'm fine if you leave it as is for now.

Reply:
Not a great practice to commit big commented out blocks of code, either :p